src/Microsoft.Health.ControlPlane.CosmosDb/Features/Storage/ControlPlaneDataStore.cs

```diff
@@ -82,6 +90,69 @@ public async Task<IEnumerable<IdentityProvider>> GetAllIdentityProvidersAsync(Ca
             return cosmosIdentityProviders.Select(cidp => cidp.ToIdentityProvider());
         }
 
+        public async Task<Role> GetRoleAsync(string name, CancellationToken cancellationToken)
+        {
+            EnsureArg.IsNotNull(name, nameof(name));
+
+            var role = await GetSystemDocumentByIdAsync<CosmosRole>(name, CosmosRole.RolePartition, cancellationToken);
+
+            if (role == null)
+            {
+                throw new RoleNotFoundException(name);
+            }
+
+            return role.ToRole();
+        }
+
+        public async Task DeleteRoleAsync(string name, CancellationToken cancellationToken)
```

Comment:
This should be able to be moved into a new switch branch under `DeleteSystemDocumentsByIdAsync`